lightning/src/ln/channel.rs

```diff
@@ -9973,7 +9970,21 @@ impl<SP: Deref> Writeable for FundedChannel<SP> where SP::Target: SignerProvider
 				},
 				&InboundHTLCState::LocalRemoved(ref removal_reason) => {
 					4u8.write(writer)?;
-					removal_reason.write(writer)?;
+					match removal_reason {
```

Comment:
Pending htlc initiated the removal (send update fail back upstream) but no response yet from peer to lock into their commit tx. See `InboundHTLCState` docs. 

This code needs to work because of resends. Fetch fail message but don't deliver. Then check if after restart message is identical. Other node just sends chan reestablish.

Reply:
Test coverage added in https://github.com/lightningdevkit/rust-lightning/pull/3645